From 0b19f0eb6d01f0b9fb28c7bb9fdef58b4b4efe1e Mon Sep 17 00:00:00 2001
From: Samul Hwang <samul_hwang@accton.com>
Date: Wed, 30 Sep 2020 09:42:44 +0200
Subject: [PATCH] ipq40xx: new NAND flash W25M02GV support for Edgecore ECW5211

This patch adds W25M02GV NAND flash support for Edgecore ECW5211

Signed-off-by: Samul Hwang <samul_hwang@accton.com>
---
 .../ipq40xx/patches-4.14/999-nand-w25m02gv.patch    | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100755 target/linux/ipq40xx/patches-4.14/999-nand-w25m02gv.patch

diff --git a/target/linux/ipq40xx/patches-4.14/999-nand-w25m02gv.patch b/target/linux/ipq40xx/patches-4.14/999-nand-w25m02gv.patch
new file mode 100755
index 0000000000..ff57048e6b
--- /dev/null
+++ b/target/linux/ipq40xx/patches-4.14/999-nand-w25m02gv.patch
@@ -0,0 +1,13 @@
+--- a/drivers/mtd/nand/nand_ids.c	2020-08-13 19:35:37.056939647 +0800
++++ b/drivers/mtd/nand/nand_ids.c	2020-08-11 17:19:36.900224383 +0800
+@@ -62,6 +62,10 @@ struct nand_flash_dev nand_flash_ids[] =
+ 		{ .id = {0xef, 0xaa} },
+ 		  SZ_2K, SZ_128, SZ_128K, NAND_NO_SUBPAGE_WRITE,
+ 		  2, 64, NAND_ECC_INFO(1, SZ_512) },
++	{"W25M02GV 256MiB 3.3V(Dual die)",
++		{ .id = {0xef, 0xab} },
++		  SZ_2K, SZ_256, SZ_128K, NAND_NO_SUBPAGE_WRITE,
++		  2, 64, NAND_ECC_INFO(1, SZ_512) },
+ 
+ 	LEGACY_ID_NAND("NAND 4MiB 5V 8-bit",   0x6B, 4, SZ_8K, SP_OPTIONS),
+ 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE3, 4, SZ_8K, SP_OPTIONS),
-- 
2.25.1

